lib/rouge/lexers/opentypefeature.rb

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+# http://www.adobe.com/devnet/opentype/afdko/topic_feature_file_syntax.html
+
+module Rouge
+  module Lexers
+    class OpenTypeFeature < RegexLexer
+      title "OpenTypeFeature"
+      desc "OpenType feature code"
+      tag 'opentypefeature'
+      aliases 'fea', 'opentype'
+      filenames '*.fea'
+
+      def self.keywords
+        @keywords ||= %w(
+          Ascender Attach CapHeight CaretOffset CodePageRange Descender FontRevision FSType
+          GlyphClassDef HorizAxis.BaseScriptList HorizAxis.BaseTagList HorizAxis.MinMax
+          IgnoreBaseGlyphs IgnoreLigatures IgnoreMarks LigatureCaretByDev LigatureCaretByIndex
+          LigatureCaretByPos LineGap MarkAttachClass MarkAttachmentType NULL Panose RightToLeft
+          TypoAscender TypoDescender TypoLineGap UnicodeRange UseMarkFilteringSet Vendor
+          VertAdvanceY VertAxis.BaseScriptList VertAxis.BaseTagList VertAxis.MinMax VertOriginY
+          VertTypoAscender VertTypoDescender VertTypoLineGap WeightClass WidthClass XHeight
+
+          anchorDef anchor anonymous anon by contour cursive device enumerate enum
+          exclude_dflt featureNames feature from ignore include_dflt include languagesystem
+          language lookupflag lookup markClass mark nameid name parameters position pos required
+          reversesub rsub script sizemenuname substitute subtable sub table useExtension
+          valueRecordDef winAscent winDescent
+        )
+      end
+
+
+      identifier = /[a-z_][a-z0-9\/_]*/i
+
+      state :root do
+        rule %r/\s+/m, Text::Whitespace
+        rule /#.*$/, Comment
+
+        # feature <tag>
+        rule /(anonymous|anon|feature|lookup|table)((?:\s)+)/ do
+          groups Keyword, Text
+          push :featurename
+        end
+        # } <tag> ;
+        rule /(\})((?:\s|\\\s)*)/ do
+          groups Punctuation, Text
+          push :featurename
+        end
+        # solve include( ../path)
+        rule /(include)/i, Keyword, :includepath
+
+        rule /[\-\[\]\/(){},.:;=%*<>']/, Punctuation
+
+        rule /`.*?/, Str::Backtick
+        rule /\"/, Str, :dqs
+
+        # classes, start with @<nameOfClass
+        rule /@#{identifier}/, Name::Class
+
+        # using negative lookbehind so we don't match property names
+        rule /(?<!\.)#{identifier}/ do |m|
+          if self.class.keywords.include? m[0]
+            token Keyword
+          else
+            token Name
+          end
+        end
+
+        rule identifier, Name
+        rule /-?\d+/, Num::Integer
+      end
+
+      state :featurename do
+        rule identifier, Name::Function, :pop!
+      end
+
+      state :includepath do
+        rule /\s+/, Text::Whitespace
+        rule /\)/, Punctuation, :pop!
+        rule /\(/, Punctuation
+        rule /[a-z0-9\/_.]*/i, Str
+      end
+
+      state :strings do
+        rule /(\([a-z0-9_]+\))?[-#0 +]*([0-9]+|[*])?(\.([0-9]+|[*]))?/i, Str::Interpol
```

Comment:
I'm a little unclear over how interpolation is supposed to work. Is there a character that indicates the beginning of the expression that should be interpolated? It looked like it was `%` before but now that's gone.

Reply:
GitHub is saying that this comment is outdated but it looks to me like line 86 is fundamentally still the same as this. There's no example of string interpolation in the sample file, is there?

Reply:
true, there is no string interpolation (variables inside a quoted string)